src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -1933,6 +1979,7 @@ codet java_bytecode_convert_methodt::convert_instructions(
     {
       assert(op.size()==2 && results.size()==1);
       results[0]=div_exprt(op[0], op[1]);
+      c=throw_arithmetic_exception(op[1], i_it->source_location);
     }
```

Comment:
This doesn't seem to be the right place for this -- this should happen during conversion from program into goto-program.